spec/triggers/aws-cloudwatch.md

```diff
@@ -0,0 +1,129 @@
+# AWS Cloudwatch Trigger
+
+This specification describes the `aws-cloudwatch` trigger for AWS Cloudwatch.
+
+```yaml
+triggers:
+  - type: aws-cloudwatch
+    authenticationRef: 
+      name: keda-trigger-auth-aws-role
+    metadata:
+      # Required: namespace
+      namespace: AWS/SQS
+      # Required: Dimension Name
+      dimensionName: QueueName
+      dimensionValue: keda
+      metricName: ApproximateNumberOfMessagesVisible
+      targetMetricValue: "2"
+      minMetricValue: "0"
+      # Required: region
+      awsRegion: "eu-west-1"    
+```
+
+- `namespace` value is the service you want to get metrics for, in this case 'AWS/SQS'. Many other options exist.
+- `dimensionName` value is the selector criteria for which resource to monitor. 
+- `dimensionValue` value is the value of the `dimensionName` you want to match.
+- `metricName` value is the metric you want to measure, these are different between namespaces.
+- `targetMetricValue` value is the average value you want to target.
+
```

Comment:
I don't want to be a pain, but the trigger spec still does not list the required authentication parameters that should be defined in the trigger authentication.

The examples below mention them but the idea of these specs is that these define what it needs to work.

Reply:
line 37 under TriggerAuthentication mentions the need for a role or iam credentials

Reply:
Since authentication requirements are part of the TriggerAuthentication I think it makes more sense to talk about it in that section instead of referencing fields of an object in a different section of the document.

Reply:
I don't think it's ideal since it only comes after the example but https://github.com/kedacore/keda/blob/master/spec/triggers/azure-storage-queue.md uses the same approach so it's fine for me.

I'll refactor it a bit once this gets merged and/or more scalers are refactored.